torch/_dynamo/variables/functions.py

```diff
@@ -1108,6 +1111,18 @@ def call_HOP(self, variable, grids, combined_args_raw, tx):
         from .constant import ConstantVariable
         from .dicts import ConstDictVariable
 
+        # as we can only pass tensors as non-const args in fx graph,
+        # here we replace TMA descriptors (TMADescriptorVariable
+        # instances) with the underlying tensors, while moving the
+        # TMA descriptor-related metadata to a separate argument,
+        # so that we can reconstruct the TMA descriptors downstream
+        tma_descriptor_metadata: TMADescriptorMetadata = {}
```

Comment:
type hint wrong?

Reply:
Not sure what you mean, could you elaborate? `TMADescriptorMetadata` is defined as

```
TMADescriptorMetadata = Dict[
    str,  # kernel parameter name
    Tuple[
        List[Union[int, SymInt]],  # dims
        List[Union[int, SymInt]],  # block_dims
        Union[int, SymInt],  # element_size
    ],
]
```

and, in the few lines below, `tma_descriptor_metadata` is filled in accordingly?

Reply:
Sorry, I didn't realize it was a dict